### PR TITLE
[homekit] add support for QR code based pairing

### PR DESCRIPTION
**Pull request by yfre**
small goodie for OH3 - support for QR code based pairing to HomeKit. 

this PR includes:
- update of java-hap library to the next version. this version  add support for qr code, linked services and new jmDNS version
- add QR code to configuration
- make use of ConfigurationAdmin in order to update Homekit configuration
- update README: pairing process using QR code and adding metadata using new UI

### CODE REVIEW

**Comment by yfre** [comment]
i have detected some high CPU load with this PR. so, converting to draft i have fixed the load issue

File: bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitImpl.java [hunk 1]
```diff
@@ -44,45 +47,93 @@
...
         Constants.SERVICE_PID + "=org.openhab.homekit", "port:Integer=9123" })
 @ConfigurableService(category = "io", label = "HomeKit Integration", description_uri = "io:homekit")
 @NonNullByDefault
 public class HomekitImpl implements Homekit {
     private final Logger logger = LoggerFactory.getLogger(HomekitImpl.class);
-    private final NetworkAddressService networkAddressService;
-    private final HomekitChangeListener changeListener;
 
-    private HomekitSettings settings;
+    @Reference
```
Comment (J-N-K) [inline bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitImpl.java:64]: Why did you change back from constructor injection to field injection?

**Review by J-N-K — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitImpl.java (hunk 1 above)
Comment (yfre) [reply bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitImpl.java:64]: i have followed the example from https://www.openhab.org/docs/developer/osgi/configadmin.html and from hueemulation..ConfigStore 
are there any recommendation?

Comment (J-N-K) [reply bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitImpl.java:64]: That needs to be updated. Thanks. The recommended way is to use constructor injection, like it was before. You can just add the `ConfigurationAdmin` as additionl parameter  to the constructor (`@Reference ConfigurationAdmin configAdmin`) and OSGi will make sure it is available before the `HomekitImpl` service starts.

File: bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitImpl.java [hunk 2]
```diff
@@ -44,45 +47,93 @@
...
+    public HomekitImpl() {
+    }
+
     @Activate
-    public HomekitImpl(@Reference StorageService storageService, @Reference ItemRegistry itemRegistry,
-            @Reference NetworkAddressService networkAddressService, Map<String, Object> config,
-            @Reference MetadataRegistry metadataRegistry) throws IOException, InvalidAlgorithmParameterException {
-        this.networkAddressService = networkAddressService;
-        this.settings = processConfig(config);
+    protected void activate(Map<String, Object> properties) {
```
Comment (J-N-K) [inline bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitImpl.java:89]:
```suggestion
    public HomekitImpl(@Reference StorageService storageService, @Reference ItemRegistry itemRegistry,
            @Reference NetworkAddressService networkAddressService, @Reference MetadataRegistry metadataRegistry,
            @Reference ConfigurationAdmin configAdmin, Map<String, Object> properties) throws IOException, InvalidAlgorithmParameterException
```

and keep the assignments to the final fields.

File: bundles/org.openhab.io.homekit/pom.xml [hunk 3]
```diff
@@ -21,9 +21,9 @@
 
   <dependencies>
     <dependency>
-      <groupId>com.github.j-n-k</groupId>
+      <groupId>io.github.yfre</groupId>
       <artifactId>hap-java</artifactId>
-      <version>2.0.0.OH2</version>
+      <version>2.1.0OH3</version>
```
[the remark this answers is not in the document]
Comment (yfre) [reply bundles/org.openhab.io.homekit/pom.xml]: done

File: bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitImpl.java (hunk 1 above)
Comment (yfre) [reply bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitImpl.java]: done

**Comment by yfre** [comment]
very sorry, looks like pushed to much
